util/util.go

```diff
@@ -410,3 +410,15 @@ func IsPaused(cluster *clusterv1.Cluster, v metav1.Object) bool {
 	_, ok := annotations[clusterv1.PausedAnnotation]
 	return ok
 }
+
+// MachinesByCreationTimestamp sorts a list of Machine by creation timestamp, using their names as a tie breaker.
+type MachinesByCreationTimestamp []clusterv1.Machine
+
+func (o MachinesByCreationTimestamp) Len() int      { return len(o) }
+func (o MachinesByCreationTimestamp) Swap(i, j int) { o[i], o[j] = o[j], o[i] }
+func (o MachinesByCreationTimestamp) Less(i, j int) bool {
+	if o[i].CreationTimestamp.Equal(&o[j].CreationTimestamp) {
+		return o[i].Name < o[j].Name
+	}
```

Comment:
\<pedantic>
Assuming the Name is unique, which it is, as long as we're comparing within one namespace, we need to compare it in order to guarantee total ordering. Without comparing by Name, the sort output will depend on in the order of inputs with equal timestamps, since `Before` returns `false` if the timestamps are equal.
\</pedantic>

(Now I'm wondering if we also need to compare on the Namespace... or the UID (but that isn't available until after the first round trip to the API) :joy:

Reply:
It might get a bit stickier than that, since we are using `sort.Sort` not `sort.StableSort`. The results wouldn't necessarily be in the same order as the inputs.

That said, for our use case and since we are going to be enforcing serial scale up, I'm not sure it really matters much. Our use of oldest is mostly just a guess at the one that should be removed.

Reply:
I'd use `sort.Slice` instead of creating yet a new type here

Reply:
We'll need this sort in upgrade as well, so I think the type is fine.

Reply:
The function can be reused. This is minor and can be fixed later